spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -565,11 +589,78 @@ def _measure_tissue_bridges(self, im_lesion_data, p_lst, idx):
     def _measure_length(self, im_data, p_lst, idx):
         """
         Measure the length of the lesion along the superior-inferior axis when taking into account the angle correction
+        The length is computed across all sagittal lesion (i.e., midsagittal and parasagittal) slices meaning that the
+        measurement is 3D.
+        For lesion length for the midsagittal slice only, see _measure_length_midsagittal_slice().
         """
         length_cur = np.sum([p_lst[2] / np.cos(self.angles_3d[zz]) for zz in np.unique(np.where(im_data)[2])])
         self.measure_pd.loc[idx, 'length [mm]'] = length_cur
         printv('  (S-I) length : ' + str(np.round(length_cur, 2)) + ' mm', self.verbose, type='info')
 
+    def _measure_length_midsagittal_slice(self, im_lesion_data, p_lst, idx):
+        """
+        Measure the length of the lesion along the superior-inferior axis in the **midsagittal slice** when taking into
+        account the angle correction.
+
+        :param im_lesion_data: 3D numpy array: mask of the lesion. The orientation is assumed to be RPI (because we
+        reoriented the image to RPI using orient2rpi())
+        :param p_lst: list, pixel size of the lesion
+        :param idx: int, index of the lesion
+        """
+        # Get the midsagittal slice number (the number is RPI oriented which is fine because we reoriented the
+        # im_lesion_data to RPI using orient2rpi())
+        # Note that this number will the same across all lesions (idx) as the midsagittal slice is based on the spinal
+        # cord segmentation
+        mid_sagittal_sc_slice = self.sagittal_sc_slice
+
+        # Get the lesion mask for the midsagittal slice
+        im_data_midsagittal = im_lesion_data[mid_sagittal_sc_slice, :, :]
+
+        # Compute the length of the lesion along the superior-inferior axis in the midsagittal slice
+        # The length is computed as the sum of the angle corrected axial slice thickness (p_lst[2])
+        # Note: p_lst[2] is the pixel size in the S-I direction
+        # Note: '[1]' in 'np.unique(np.where(im_data_midsagittal)[1])' is used to get the unique axial slices (S-I
+        # direction) with the lesion
+        length_cur = np.sum([p_lst[2] / np.cos(self.angles_sagittal[axial_slice])
+                             for axial_slice in np.unique(np.where(im_data_midsagittal)[1])])
```

Comment:
I think this is the right formula for angle correction of the lesion length in the mid-sagittal slice. (Axial slice thickness _divided_ by cos of the angle in the sagittal plane.)